tests/testsuite/cargo_command.rs

```diff
@@ -477,16 +517,43 @@ fn cargo_subcommand_args() {
 
     cargo_process("echo bar -v --help")
         .env("PATH", &path)
-        .with_stdout("echo bar -v --help")
+        .with_stdout_data(str![[r#"
+echo bar -v --help
+
+"#]])
         .run();
 }
 
 #[cargo_test]
 fn explain() {
     cargo_process("--explain E0001")
-        .with_stdout_contains(
-            "This error suggests that the expression arm corresponding to the noted pattern",
-        )
+        .with_stdout_data(str![[r#"
+#### Note: this error code is no longer emitted by the compiler.
+
+This error suggests that the expression arm corresponding to the noted pattern
+will never be reached as for all possible values of the expression being
+matched, one of the preceding patterns will match.
+
+This means that perhaps some of the preceding patterns are too general, this
+one is too specific or the ordering is incorrect.
+
+For example, the following `match` block has too many arms:
+
+```
+match Some(0) {
+    Some(bar) => {/* ... */}
+    x => {/* ... */} // This handles the `None` case
+    _ => {/* ... */} // All possible cases have already been handled
+}
+```
+
+`match` blocks have their patterns matched in order, so, for example, putting
+a wildcard arm above a more specific arm will make the latter arm irrelevant.
+
+Ensure the ordering of the match arm is correct and remove any superfluous
+arms.
+
```

Comment:
The error is from rustc and subject to change. 

```suggestion
...
[..]This error suggests that the expression arm corresponding to the noted pattern[..]
...
```

Reply:
modified